File: trunk/ShadowFrameworkLite20/src/shadow/geometry/functions/SFRectangle2DFunction.java
package shadow.geometry.functions;


public class SFRectangle2DFunction  extends SFUnoptimizedSurfaceFunction{

	private float x,y,w,h;

	public SFRectangle2DFunction(float x, float y, float w, float h) {
		super();
		this.x = x;
		this.y = y;
		this.w = w;
		this.h = h;
	}
	
	@Override
	public float getX(float u, float v) {
		return x+u*w;
	}
	
	@Override
	public float getY(float u, float v) {
		return y+v*h;
	}
	
	@Override
	public float getZ(float u, float v) {
		return 0;
	}

	
	@Override
	public void init() {
		//Do nothing
	}
	
	@Override
	public void destroy() {
		//Its correct: if init isn't doing anything, destroy should not do anything
	}

	public float getX() {
		return x;
	}

	public void setX(float x) {
		this.x = x;
	}

	public float getY() {
		return y;
	}

	public void setY(float y) {
		this.y = y;
	}

	public float getW() {
		return w;
	}

	public void setW(float w) {
		this.w = w;
	}

	public float getH() {
		return h;
	}

	public void setH(float h) {
		this.h = h;
	}
	
	
}

File: trunk/ShadowFrameworkJS/src/shadow/geometry/functions/SFRectangle2DFunction.js

function SFRectangle2DFunction(x,y,w,h){
	
		this.x = x;
		this.y = y;
		this.w = w;
		this.h = h;
}

inherit(SFRectangle2DFunction,SFUnoptimizedSurfaceFunction);

SFRectangle2DFunction.prototype["getX"]=function(u,v){
	return this.x+u*this.w;
};

SFRectangle2DFunction.prototype["getY"]=function(u,v){
	return this.y+u*this.h;
};

SFRectangle2DFunction.prototype["getZ"]=function(u,v){
	return 0;
};
	
SFRectangle2DFunction.prototype["init"]=function(){
	//do nothing
};

SFRectangle2DFunction.prototype["destroy"]=function(){
	//do nothing
};
	
	
SFRectangle2DFunction.prototype["getX"]=function(){
	return this.x;
};
	
SFRectangle2DFunction.prototype["getY"]=function(){
	return this.y;
};

SFRectangle2DFunction.prototype["getW"]=function(){
	return this.w;
};

SFRectangle2DFunction.prototype["getH"]=function(){
	return this.h;
};


SFRectangle2DFunction.prototype["setX"]=function(x){
	this.x=x;
};

SFRectangle2DFunction.prototype["setY"]=function(y){
	this.y=y;
};

SFRectangle2DFunction.prototype["setW"]=function(w){
	this.w=w;
};

SFRectangle2DFunction.prototype["setH"]=function(h){
	this.h=h;
};
